state/protocol/badger/state.go

```diff
@@ -236,14 +237,30 @@ func (state *State) bootstrapStatePointers(root protocol.Snapshot) func(*badger.
 		highest := segment.Highest()
 		lowest := segment.Lowest()
 
+		safetyData := &hotstuff.SafetyData{
+			LockedOneChainView:      highest.Header.View,
+			HighestAcknowledgedView: highest.Header.View,
+		}
+
+		rootQC, err := root.QuorumCertificate()
+		if err != nil {
+			return fmt.Errorf("could not get root QC: %w", err)
+		}
+
+		livenessData := &hotstuff.LivenessData{
+			CurrentView: highest.Header.View + 1,
+			LastViewTC:  highest.Header.LastViewTC,
```

Comment:
I don't understand why this would be a correct initialization. Reasoning:
* Let `V` denote the view of the `highest` block in the sealing segment (i.e. `V := highest.Header.View`)
* If `highest.Header.LastViewTC` is set, it represents the TC for the view `V-1` 
* We specify that a consensus node should start at round `V+1`: https://github.com/onflow/flow-go/blob/af9fc5c3cdca409ac82457bf4237f250b46f96d6/state/protocol/badger/state.go#L251 So a valid TC for the "prior round" should have view `V`

But from my perspective, we can simply set `LastViewTC` to nil. Based on my previous comment, `rootQC`  should be for `highest` block. As the replica knows a QC for view `V`, it can enter view `V+1` according to consensus rules. No TC is required. 
```suggestion
```

Reply:
agree, good reasoning.